### PR TITLE
Use IColonyMetadata to render colonies.

### PR DESCRIPTION
**Pull request by kberg**
Colonies are all hard-coded. This will help.

### CODE REVIEW

File: src/client/components/colonies/ColonyTradeRow.vue [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+      <div><div class="resource money red-outline"></div></div>
+      <div><div class="resource heat red-outline"></div></div>
+      <div><div class="resource energy red-outline"></div></div>
+      <div><div class="resource plant red-outline"></div></div>
+      <div><div class="resource steel red-outline"></div></div>
+      <div><div class="resource titanium red-outline"></div></div>
+    </template>
+    <template v-else-if="metadata.name === ColonyName.EUROPA">
+    </template>
+    <template v-else v-for="idx in range(7)">
```
Comment (Argeento) [inline src/client/components/colonies/ColonyTradeRow.vue:23]:
```suggestion
    <template v-else v-for="idx in 7">
```

File: src/client/components/colonies/ColonyTradeRow.vue [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+      <div :key="idx">{{metadata.tradeQuantity[idx]}}</div>
```
Comment (Argeento) [inline src/client/components/colonies/ColonyTradeRow.vue:24]:
```suggestion
      <div :key="idx">{{metadata.tradeQuantity[idx -1]}}</div>
```

File: src/client/components/colonies/ColonyTradeRow.vue [hunk 3]
```diff
@@ -0,0 +1,55 @@
...
+
+export default Vue.extend({
+  name: 'ColonyTradeRow',
+  props: {
+    metadata: {
+      type: Object as () => IColonyMetadata,
+      required: true,
+    },
+  },
+  computed: {
```
Comment (Argeento) [inline src/client/components/colonies/ColonyTradeRow.vue:44]:
```suggestion
  data() {
    return {
       ColonyName
    }
```

File: src/client/components/colonies/ColonyRow.vue [hunk 4]
```diff
@@ -0,0 +1,35 @@
...
+export default Vue.extend({
+  components: {BuildBenefit},
+  name: 'ColonyRow',
+  props: {
+    metadata: {
+      type: Object as () => IColonyMetadata,
+      required: true,
+    },
+  },
+  computed: {
```
Comment (Argeento) [inline src/client/components/colonies/ColonyRow.vue:29]:
```suggestion
  data() {
    return {
       ColonyName
    }
```

File: src/client/components/colonies/BuildBenefit.vue [hunk 5]
```diff
@@ -0,0 +1,79 @@
...
+  },
+  computed: {
+    ColonyBenefit(): typeof ColonyBenefit {
+      return ColonyBenefit;
+    },
+    buildQuantityText(): string {
+      return this.metadata.buildQuantity[this.idx] > 1 ? String(this.metadata.buildQuantity[this.idx]) : '';
+    },
+    resource(): string | undefined {
+      console.log(this.metadata.name, this.metadata.buildResource, this.metadata.resourceType);
```
Comment (Argeento) [inline src/client/components/colonies/BuildBenefit.vue:74]: Unnecessary console.log

File: src/client/components/LogPanel.vue [hunk 6]
```diff
@@ -316,22 +315,8 @@ export default Vue.extend({
...
-          current: globalEvent.currentDelegate,
-        };
-      }
-      return {
-        name: globalEventName,
-        description: 'global event not found',
-        revealed: PartyName.GREENS,
-        current: PartyName.GREENS,
-      };
+    getGlobalEventModel(globalEventName: GlobalEventName): GlobalEventModel {
```
Comment (Argeento) [inline src/client/components/LogPanel.vue:318]:
```suggestion
    getGlobalEventByName,
```

File: src/client/components/DebugUI.vue [hunk 7]
```diff
@@ -388,29 +387,13 @@ export default Vue.extend({
...
-          current: globalEvent.currentDelegate,
-        };
-      }
-      return {
-        name: globalEventName,
-        description: 'global event not found',
-        revealed: PartyName.GREENS,
-        current: PartyName.GREENS,
-      };
+    getGlobalEventModel(globalEventName: GlobalEventName): GlobalEventModel {
```
Comment (Argeento) [inline src/client/components/DebugUI.vue:395]:
```suggestion
    getGlobalEventByName,
```

File: src/client/components/colonies/BuildBenefit.vue (hunk 5 above)
Comment (bafolts) [reply src/client/components/colonies/BuildBenefit.vue:74]: Remove before flight.

**Review by bafolts — APPROVED** [review]

**Pull request merged**